packages/api-docs-builder/ApiBuilders/HookApiBuilder.ts

```diff
@@ -0,0 +1,539 @@
+            // const Component = () => {}
+            node = binding.path.parentPath!.node;
+          }
+        }
+      }
+
+      const { leadingComments } = node;
+      const leadingCommentBlocks =
+        leadingComments != null
+          ? leadingComments.filter(({ type }) => type === 'CommentBlock')
+          : null;
+      const jsdocBlock = leadingCommentBlocks != null ? leadingCommentBlocks[0] : null;
+      if (leadingCommentBlocks != null && leadingCommentBlocks.length > 1) {
+        throw new Error(
+          `Should only have a single leading jsdoc block but got ${
+            leadingCommentBlocks.length
+          }:\n${leadingCommentBlocks
+            .map(({ type, value }, index) => `#${index} (${type}): ${value}`)
+            .join('\n')}`,
+        );
+      }
+      if (jsdocBlock?.start != null && jsdocBlock?.end != null) {
+        start = jsdocBlock.start;
+        end = jsdocBlock.end;
+      } else if (node.start != null) {
+        start = node.start - 1;
+        end = start;
+      }
+    },
+  });
+
+  if (end === null || start === 0) {
+    throw new TypeError(
+      "Don't know where to insert the jsdoc block. Probably no `default export` found",
+    );
+  }
+
+  const markdownLines = (await computeApiDescription(api, { host: HOST })).split('\n');
+  // Ensure a newline between manual and generated description.
+  if (markdownLines[markdownLines.length - 1] !== '') {
+    markdownLines.push('');
+  }
+
+  if (api.demos && api.demos.length > 0) {
+    markdownLines.push(
+      'Demos:',
+      '',
+      ...api.demos.map((item) => {
+        return `- [${item.name}](${
+          item.demoPathname.startsWith('http') ? item.demoPathname : `${HOST}${item.demoPathname}`
+        })`;
+      }),
+      '',
+    );
+  }
+
+  markdownLines.push(
+    'API:',
+    '',
+    `- [${api.name} API](${
+      api.apiPathname.startsWith('http') ? api.apiPathname : `${HOST}${api.apiPathname}`
+    })`,
+  );
+
+  const jsdoc = `/**\n${markdownLines
+    .map((line) => (line.length > 0 ? ` * ${line}` : ` *`))
+    .join('\n')}\n */`;
+  const typesSourceNew = typesSource.slice(0, start) + jsdoc + typesSource.slice(end);
+  writeFileSync(typesFilename, typesSourceNew, { encoding: 'utf8' });
+}
+
+const attachTable = (
+  reactApi: ReactApi,
+  params: ParsedProperty[],
+  tableName: 'parametersTable' | 'returnValueTable',
+) => {
+  const propErrors: Array<[propName: string, error: Error]> = [];
+  const parameters: ReactApi[typeof tableName] = params
+    .map((p) => {
+      const { name: propName, ...propDescriptor } = p;
+      let prop: Omit<ParsedProperty, 'name'> | null;
+      try {
+        prop = propDescriptor;
+      } catch (error) {
+        propErrors.push([propName, error as Error]);
+        prop = null;
+      }
+      if (prop === null) {
+        // have to delete `componentProps.undefined` later
+        return [] as any;
+      }
+
+      // Only keep `default` for bool props if it isn't 'false'.
```

Comment:
There's no harm for us in being explicit and stating all boolean props' default values, not just non-false ones.

Reply:
I followed here the logic in the components' props builder. We could fix it separately in both scripts. There was already an issue related to this - https://github.com/mui/material-ui/issues/35822